ui/src/app/views/registration-dashboard.tsx

```diff
@@ -110,7 +113,7 @@ export const RegistrationTasks = [
       return !!profile.eraCommonsCompletionTime || !!profile.eraCommonsBypassTime;
     },
     onClick: redirectToNiH
-  }, {
+  }, ...(serverConfigStore.getValue().enableDataUseAgreement ? [{
```

Comment:
I don't think its that necessary but I thought of a cute trick that looks cleaner to me. We can add another property to each task called `isEnabled` and we can filter on that value to return the correct list of tasks.

The default can be `() => true` but for values backed by serverConfig, we can do the API call.